guardian/editorial-collaboration
# Create Hello World express app

Creates the skeleton for a Node app. This will become the service which receives prosemirror-collab events ('steps').

Technologies used:

* node (20.13.1)
* npm 
* express
* Typescript (ts-node, tsc, ts-config)
* webpack
* GuCDK (with riff-raff.yaml generation)

The domain name is defined in GuCDK. The UserData statement sets up a SystemD service[^1].

This service uses a [new Node 20 AMIgo image](https://amigo.gutools.co.uk/recipes/editorial-tools-focal-node20-ARM), which implements the [CDK Base role](https://amigo.gutools.co.uk/roles#cdk-base)[^2]. 

### How to test
At the time of writing I have deployed this to CODE: https://editorial-collaboration.code.dev-gutools.co.uk/. You should be able to see 'Hello World!'.

You can also deploy this branch to CODE yourself, or test the webpack bundling and express server locally.

[^1]: Note that in Scala projects this is handled for us by Debian packaging.
[^2]: This allows us to: (1) fetch instance tags and store under /etc/config (via [instance-tag-discovery](https://github.com/guardian/instance-tag-discovery)), (2) ship cloud-init-output and application logs to a Kinesis stream (via [devx-logs](https://github.com/guardian/devx-logs)).
